packages/backend/src/core/collectors/SpotValidiumStateTransitionCollector.ts

```diff
@@ -0,0 +1,74 @@
+import { Hash256 } from '@explorer/types'
+import { EthereumAddress } from '@explorer/types/src/EthereumAddress'
+
+import { BlockRange } from '../../model/BlockRange'
+import {
+  StateTransitionRecord,
+  StateTransitionRepository,
+} from '../../peripherals/database/StateTransitionRepository'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { BlockNumber } from '../../peripherals/ethereum/types'
+import { ValidiumStateTransition } from '../PerpetualValidiumUpdater'
+import { LogRootUpdate, LogStateTransitionFact } from './events'
+
+export class SpotValidiumStateTransitionCollector {
```

Comment:
The diff between this and PerpetualValidiumStateTransitionCollector is soooo small. I'm torn between letting this go because technically they are something different and recommending having a common superclass or extracting a common function for processing the events.